editor/editor_data.cpp

```diff
@@ -721,8 +721,11 @@ bool EditorData::check_and_update_scene(int p_idx) {
 		}
 
 		new_scene->set_scene_file_path(edited_scene[p_idx].root->get_scene_file_path());
-
-		memdelete(edited_scene[p_idx].root);
+		Node *old_root = edited_scene[p_idx].root;
+		for (int i = 0; i < old_root->get_child_count(); i++) {
+			memdelete(old_root->get_child(i));
+		}
+		old_root->replace_by(new_scene);
```

Comment:
`replace_by()` does not delete the node, so unless it's still referenced somewhere, this will cause a leak.

Reply:
yeah you are right I forgot to delete it as I always forget that it does not delete the node :(